tests/testResolutionsClass.m

```diff
@@ -0,0 +1,283 @@
+            testCase.verifyEqual(resolution.resolutions.typesTable{1, 1}, "Resolution 1", 'setResolutionValue method not working');
+
+            resolution.setResolutionValue(1, 'type', 'constant');
+            testCase.verifyEqual(resolution.resolutions.typesTable{1, 2}, "constant", 'setResolutionValue method not working');
+            resolution.setResolutionValue('Resolution 1', 2, 'gaussian'); 
+            testCase.verifyEqual(resolution.resolutions.typesTable{1, 2}, "gaussian", 'setResolutionValue method not working');
+            testCase.verifyError(@() resolution.setResolutionValue(2, 'type', 'random'), invalidOption.errorID);
+            
+            resolution.setResolutionValue(2, 'Value 1', 'random');
+            testCase.verifyEqual(resolution.resolutions.typesTable{2, 3}, "random", 'setResolutionValue method not working');
+            resolution.setResolutionValue('Resolution 2', 3, 'Resolution par 1');
+            testCase.verifyEqual(resolution.resolutions.typesTable{2, 3}, "Resolution par 1", 'setResolutionValue method not working');
+            
+            resolution.setResolutionValue(2, 'Value 2', 'Resolution par 1');
+            testCase.verifyEqual(resolution.resolutions.typesTable{2, 4}, "Resolution par 1", 'setResolutionValue method not working');
+            testCase.verifyError(@() resolution.setResolutionValue(2, 'Value 2', 'random'), nameNotRecognised.errorID);
+         end
+
+         function testDisplayResolutionsObject(testCase)
+            import matlab.unittest.constraints.ContainsSubstring
+            % Check that the contents of the resolution are printed
+            paramHeader = {'p', 'Name', 'Min', 'Value', 'Max', 'Fit?'};
+            resolution = resolutionsClass(testCase.parameters(1, :), testCase.resolutions(1, :));
+            
+            display = evalc('resolution.displayResolutionsObject()');
+            testCase.verifyNotEmpty(display, 'displayResolutionsObject method not working');
+            display = split(display, ["(a)", "(b)"]);  % splits resolution parameter table from resolution table
+            testCase.verifyLength(display, 3);
+
+            displayArray = textscan(display{2},'%s','Delimiter','\r','TextType','string');
+            displayArray = strip(displayArray{1});
+            % display table should be height of resolPars table plus header and divider row
+            testCase.verifyLength(displayArray, height(resolution.resolPars.paramsTable) + 3);
+            % Remove html tags used to format header then split table when
+            % 2 or more spaces are found to avoid splitting names with single space
+            displayHeader = eraseBetween(displayArray{2}, '<', '>', 'Boundaries','inclusive');
+            displayHeader = regexp(displayHeader, '\s{2,}', 'split');
+            testCase.verifyEqual(displayHeader, paramHeader);
+	        row = regexp(displayArray{4}, '\s{2,}', 'split');
```

Comment:
Why is this line (and others below) indented?

Reply:
I have no idea, it doesn't show up in the MATLAB editor but shows up in vscode. Now fixed